device/alpaka/src/seeding/spacepoint_formation_algorithm.cpp

```diff
@@ -67,13 +70,15 @@ spacepoint_formation_algorithm::operator()(
     auto workDiv = makeWorkDiv<Acc>(blockSize, nBlocks);
 
     // Launch the spacepoint formation kernel.
-    ::alpaka::exec<Acc>(queue, workDiv, FormSpacepointsKernel{},
-                        measurements_view, det_descr, num_measurements,
+    ::alpaka::exec<Acc>(queue, workDiv, FormSpacepointsKernel<detector_t>{},
+                        det_view, measurements_view, num_measurements,
                         vecmem::get_data(spacepoints));
     ::alpaka::wait(queue);
 
     // Return the reconstructed spacepoints.
     return spacepoints;
 }
+// Explicit template instantiation
+template class spacepoint_formation_algorithm<default_detector::device>;
```

Comment:
:smile: I'm very happy that you made use of the typedefs that I added a while ago. As this was exactly my idea with them.

However, while this way of writing the code does work, I think the user interface provided like this is not the easiest to understand. :frowning: I think we should provide explicit typedefs in the header files for the template specializations that we would actually provide in the different libraries. I.e. I'd put something like the following into the algorithm headers:

```c++
using default_spacepoint_formation_algorithm = spacepoint_formation_algorithm<default_detector::device>;
```

So that as long as the user only used existing typedefs, they could be sure that the algorithm would be available in the library. :thinking:

Reply:
Do you mean making a new header file for these kind of typedefs?
```
using default_spacepoint_formation_algorithm = spacepoint_formation_algorithm<default_detector::device>;
using default_finding_algorithm = finding_algorithm<default_stepper_type, default_navigator_type>;
... etc.
```

I like the idea but let's do this in another PR. Doing this for cpu, cuda, sycl, and alpaka is already a huge task :skull:

Reply:
No, not a separate header.

I'm trying to figure out the best setup myself as well... :thinking: I have an idea right now, but it will need some experimentation. So, let's just go with this type of templating for now, and I'll propose a slightly different code layout in a follow-up.